backend/app/db/weaviate/user_profiling.py

```
@@ -0,0 +1,310 @@
+        """
+        Analyze language frequency across repositories to identify top 5 languages.
+        """
+        language_counter = Counter()
+        for repo in repositories:
+            language_counter.update(repo.languages)
+
+        top_languages = language_counter.most_common(5)
+        logger.info(f"Top 5 languages by frequency: {top_languages}")
+        return [lang for lang, _ in top_languages]
+
+    async def build_user_profile(self, user_id: str, github_username: str) -> Optional[WeaviateUserProfile]:
```

Pylint (3.3.7)

[refactor] Too many local variables (22/15)

(R0914)